### ISSUE TITLE
Disabled elements should be interactable for a11y

### ISSUE THREAD

**Issue by mayank99** [iTwin/iTwinUI#391]
> when a button is disabled, it can no longer show a tooltip on hover.  I think if we offer a class name that makes a button appear disabled, we should be able to show a tooltip on hover / focus.

_Originally posted by @FlyersPh9 in https://github.com/iTwin/iTwinUI/pull/363#discussion_r744888686_

**Comment by mayank99** [comment]
Another use for `aria-disabled` is when showing a loading spinner inside the button.

https://user-images.githubusercontent.com/9084735/224162889-38720f1f-5c20-46a3-af45-d0dd60ba147a.mov

When loading, the button will become disabled. The focus should stay on the button for better accessibility. This will be useful when the button is used to open a dialog (the focus will return to button when dialog closes) and also if the loading fails, keyboard users can easily press <kbd>Enter</kbd> again.

**Comment by FlyersPh9** [comment]
It looks like [the button CSS already has `aria-disabled`](https://github.com/iTwin/iTwinUI/blob/15c480d7cb321d1dd13b5359581c1c888a733f81/packages/itwinui-css/src/button/button.scss#L61), so do we just need to add support for this in React?

**Comment by mayank99** [comment]
yes, we need to manually prevent clicks when using aria-disabled
